### ISSUE TITLE
`Meta.meta` returns a broken value for Types

### ISSUE THREAD

**Issue by radeusgd** [enso-org/enso#5805]
See the following example:
```
from Standard.Base import all

type My_Type
    My_Ctor x

main =
    v1 = My_Type.My_Ctor 42
    IO.println (Meta.is_atom v1)
    m1 = Meta.meta v1
    IO.println m1
    IO.println m1.constructor.fields

    v2 = My_Type
    IO.println (Meta.is_atom v2)
    m2 = Meta.meta v2
    case m2 of
        Meta.Atom.Value _ ->
            IO.println m2
            IO.println m2.constructor.fields
```

### Expected behaviour

- A) it should print:
```
True
(Atom.Value (My_Ctor 42))
[x]
True
(Atom.Value My_Type)
[]
```

- B) it should print sth like:
```
True
(Atom.Value (My_Ctor 42))
[x]
False
(Type.Value My_Type)
```

The variant **B** is my personal preference - I'm not sure if `My_Type` is and should be an Atom? I think we should introduce a separate `Meta.Type` to hold the types. This would make our life easier also in regards to #5792, allowing us to check if the second `is_a` argument is a `Type` and reject invalid arguments easily.

### Current behaviour

```
True
(Atom.Value (My_Ctor 42))
[x]
True
(Atom.Value My_Type)
Execution finished with an error: Unsupported argument for arg1 expected a AtomConstructor
        at <enso> Meta.get_constructor_fields(Internal)
        at <enso> Constructor.fields<arg-1>(C:\NBO\enso\built-distribution\enso-engine-0.0.0-dev-windows-amd64\enso-0.0.0-dev\lib\Standard\Base\0.0.0-dev\src\Meta.enso:64:37-60)
        at <enso> Constructor.fields(C:\NBO\enso\built-distribution\enso-engine-0.0.0-dev-windows-amd64\enso-0.0.0-dev\lib\Standard\Base\0.0.0-dev\src\Meta.enso:64:9-61)
        at <enso> case_branch<arg-1>(metatype.enso:19:24-44)
        at <enso> case_branch(metatype.enso:19:13-44)
        at <enso> metatype.main(metatype.enso:16-19)
```

The `fields` method crashes. This is definitely not desirable.

I'd suggest to fix this by going (B) and not making types Atoms if that is OK. But a fix that will go the (A) route sounds acceptable too. We just shouldn't crash.

**Comment by JaroslavTulach** [comment]
`Type` is [treated as atom in IsAtomNode](https://github.com/enso-org/enso/blame/7b0759f8b3b7ab4276a8668642dbd4b9586fafdd/engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/IsAtomNode.java#L16) - written by @kustosz.

**Comment by JaroslavTulach** [comment]
Variant A seems as the simplest fix: please take a look at #5869.

**Comment by radeusgd** [comment]
<img width="941" alt="image" src="https://user-images.githubusercontent.com/1436948/224666564-1772a1ff-10dc-49a2-9ec5-068690d88138.png">
I've also found out the same issue happens for Modules which are also not really atoms but are treated as such.

Does the fix also apply there? Can we add a test case for it?
